src/app/libp2p_helper/src/codanet.go

```diff
@@ -539,6 +559,39 @@ func (h *Helper) handleNodeStatusStreams(s network.Stream) {
 	logger.Debugf("wrote node status to stream %s", s.Protocol())
 }
 
+func (h *Helper) handleHandshakeStream(s network.Stream) {
+	defer func() {
+		err := s.Close()
+		if err != nil {
+			logger.Error("failed to close write side of stream", err)
+			return
+		}
+
+		<-time.After(400 * time.Millisecond)
+
+		err = s.Reset()
+		if err != nil {
+			logger.Error("failed to reset stream", err)
+		}
+	}()
+
+	data, err := json.Marshal(h.IsConsensusNode)
+	if err != nil {
+		logger.Error("failed to unmarshal Handshake data", err)
```

Comment:
failed to marshal*
also we probably want to return after this

Reply:
Done